api_client.go

```diff
@@ -246,9 +246,11 @@ func NewAPIClient(cfg *Configuration) (*APIClient, error) {
 	c.cfg = cfg
 	c.Context = GetContext(cfg)
 	c.common.client = c
-	err := GetDefaultHeaders(c)
-	if err != nil {
-		return nil, err
+	if !c.cfg.SkipSessionAuth {
+		err := GetDefaultHeaders(c)
```

Comment:
Why only session-based auth need default header? I am asking because we also need this function previously.